### PR TITLE
Improve cache search

### PR DESCRIPTION
**Pull request by jerelmiller**
Searching the cache in the devtools has always felt a little bit broken. At present, it doesn't do any filtering of the list, so matches against the search string are very difficult to find if the list of cache keys is large (see example in [this discussion](https://github.com/apollographql/apollo-client-devtools/discussions/867)). Furthermore, it would match against values in the cache entry itself, but this was difficult to find since it required you to activate the cache entry to see it.

This PR improves the search functionality by matching only against cache keys and actively filtering that list as the user types. The matched string in the cache key is also now highlighted so its obvious how the record is filtered. This should improve cases where you might have an ID and want to quickly see what what a record with that ID contains by pasting that ID into the search field.

**Before**
search-old

**After**
search-new

### CODE REVIEW

**Review by alessbell — APPROVED** [review]
This is such a great usability improvement! 🚀

**Comment by jerelmiller** [comment]
Will merge once #1052 is merged since this branches off of that 🙂

**Comment by alessbell** [comment]
Just noticed that :) Was hoping I could sneak this into the next release I'm about to cut but can't review the other one before signing off today 😅

**Pull request merged**
